test/mail_dhparams_default.bats

```diff
@@ -0,0 +1,67 @@
+load 'test_helper/common'
+
+function setup() {
+    run_setup_file_if_necessary
+}
+
+function teardown() {
+    run_teardown_file_if_necessary
+}
+
+function setup_file() {
+    docker run -d --name mail_default_dhparams_one_dir \
+		-v "`pwd`/test/config":/tmp/docker-mailserver \
+		-v "`pwd`/test/test-files":/tmp/docker-mailserver-test:ro \
+		-e DMS_DEBUG=0 \
+		-e ONE_DIR=1 \
+		-h mail.my-domain.com -t ${NAME}
+
+    docker run -d --name mail_default_dhparams_not_one_dir \
+		-v "`pwd`/test/config":/tmp/docker-mailserver \
+		-v "`pwd`/test/test-files":/tmp/docker-mailserver-test:ro \
+		-e DMS_DEBUG=0 \
+		-e ONE_DIR=0 \
+		-h mail.my-domain.com -t ${NAME}
+
+    wait_for_finished_setup_in_container mail_default_dhparams_one_dir
+    wait_for_finished_setup_in_container mail_default_dhparams_not_one_dir
+}
```

Comment:
> I'm not sure where it should be, but I think a clearer description that differentiates what this test vs the other two separate ones cover would be useful.

Agreed.

> As you're using volume mounts with the same local location for test/config:/tmp/docker-mailserver with no :ro flag, it suggests they'll both be writing data to this location. Does that risk contaminating the test if both containers try to write a file at the same time to the location? I understand your PR doesn't, but I'm not sure if that is reliable to assume nothing else will in future. May be best to provide separate local locations to volume mount to avoid that?

Read-only storage make sense, but I've already experienced some permissions issues during production using docker configs (which are  intrinsically read-only) (https://github.com/tomav/docker-mailserver/blob/master/target/start-mailserver.sh#L1578) .
I'll see if no warning is raised when using the `:ro` flag, none of the others tests are not using it, see https://github.com/tomav/docker-mailserver/blob/master/Makefile#L27.

It could be an improvement to have separated config folders; one per container, instead of sharing data between containers, but that would require a lot of refactoring... ;(

We could open an issue for setting read-only `test/config` in tests when writing operations are not required.

Reply:
> I'll see if no warning is raised when using the :ro flag, none of the others tests are not using it, see https://github.com/tomav/docker-mailserver/blob/master/Makefile#L27.

As stated, I am not familiar with how tests are done in this project. Perhaps writing to the location occurs while running the image elsewhere in the code, if the others aren't using `:ro`, then this one shouldn't either.

> It could be an improvement to have separated config folders; one per container, instead of sharing data between containers, but that would require a lot of refactoring... ;(

Is there a requirement for the test to cover both `ONE_DIR` conditions, or would the problem be solved splitting it into two files like the other two?(probably not DRY, but it's already being spread over 3 tests)

The decision can be left up to @erik-wramner or anyone else who will approve and merge the PR, I am just concerned about running those two in parallel if they expect to share the same local storage as `:rw`. The link you provided shows a `sleep 15` between container runs, which is presumably enough time for the test to run and then exit, which should not cause any problems.

Reply:
>  I'll see if no warning is raised when using the :ro flag, none of the others tests are not using it, see https://github.com/tomav/docker-mailserver/blob/master/Makefile#L27.

 ```
/usr/local/bin/start-mailserver.sh: line 900: /tmp/docker-mailserver/postfix-aliases.cf: Read-only file system
```
A warning was raised during the tests. Read-only refactoring should be treated in a different PR/issue.

Reply:
> The decision can be left up to @erik-wramner or anyone else who will approve and merge the PR, I am just concerned about running those two in parallel if they expect to share the same local storage as :rw. The link you provided shows a sleep 15 between container runs, which is presumably enough time for the test to run and then exit, which should not cause any problems.

The `wait_for_finished_setup_in_container` instruction has been added after the `docker run` instruction. Containers are now setting up sequentially.
Thank you for having highlighted this, nice catch!